test/jit/test_device_analysis.py

```diff
@@ -173,3 +173,23 @@ def test_fn(x, y, z: bool):
         self.assert_device_equal(test_fn, [self.cpu, self.cpu, None], self.cpu)
         self.assert_device_equal(test_fn, [self.mkldnn, self.mkldnn, None], self.mkldnn)
         self.assert_device_equal(test_fn, [self.cpu, self.cuda, None], None)
+
+    def test_loop_simple(self):
+        def test_fn(x, y, z: int):
+            for _ in range(z):
+                y = x
+            return y
+
+        self.assert_device_equal(test_fn, [self.cpu, self.cpu, None], self.cpu)
+        self.assert_device_equal(test_fn, [self.cpu, self.cuda, None], None)
+        self.assert_device_equal(test_fn, [self.cpu, None, None], None)
+
+    def test_loop_device_change(self):
```

Comment:
could you please add a few more tests with nested loops and a while loop and maybe with an if statement?